wazuh/wazuh-dashboard-plugins
# Use index pattern id instead of title

### Description
This PR resolves the issue where dashboards do not filter correctly when selecting an index pattern with an ID different from its name. The solution involves updating the application to use the index pattern ID instead of the title.

#### Changes
- Modified the application to use the index pattern ID for filtering in dashboards.
 
### Issues Resolved
https://github.com/wazuh/wazuh-dashboard-plugins/issues/6742

### Evidence

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/e9f6a216-576e-451f-a99d-f4e9302734e7)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/a85cecef-01a8-4a44-842a-5f8b16f35da2)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/8076766f-2d26-47c9-a0d3-c8f99508b511)
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/103193307/f804a920-699a-4511-ba27-a2332d8ef1d8)

### Test
1. Create an index pattern with an ID that differs from its name.
2. Navigate to the top bar selector and select the newly created index pattern.
3. Verify the following cases:

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Threat Hunting Dashboard tab -> Check fixed filters and data are consistent | :black_circle: | :black_circle: | :black_circle: |
| Threat Hunting Events tab -> Check fixed filters and data are consistent | :black_circle: | :black_circle: | :black_circle: |
| Generate PDF report from Threat Hunting | :black_circle: | :black_circle: | :black_circle: |
| Vulnerability Detection Dashboard tab -> Check fixed filters and data are consistent | :black_circle: | :black_circle: | :black_circle: |
| Vulnerability Detection Inventory tab -> Check fixed filters and data are consistent | :black_circle: | :black_circle: | :black_circle: |
| Vulnerability Detection Events tab -> Check fixed filters and data are consistent | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Threat Hunting Dashboard tab -> Check fixed filters and data are consistent</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Threat Hunting Events tab -> Check fixed filters and data are consistent</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate PDF report from Threat Hunting</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Vulnerability Detection Dashboard tab -> Check fixed filters and data are consistent</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Vulnerability Detection Inventory tab -> Check fixed filters and data are consistent</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Vulnerability Detection Events tab -> Check fixed filters and data are consistent</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
